gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -0,0 +1,331 @@
+   * A builder for {@code ChangeRequest}s.
+   */
+  public static class Builder {
+
+    private List<DnsRecord> additions = new LinkedList<>();
+    private List<DnsRecord> deletions = new LinkedList<>();
+    private String id;
+    private Long startTimeMillis;
+    private Status status;
+
+    private Builder(ChangeRequest cr) {
+      this.additions = Lists.newLinkedList(cr.additions());
+      this.deletions = Lists.newLinkedList(cr.deletions());
+      this.id = cr.id();
+      this.startTimeMillis = cr.startTimeMillis();
+      this.status = cr.status();
+    }
+
+    private Builder() {
+    }
+
+    /**
+     * Sets a collection of {@link DnsRecord}s which are to be added to the zone upon executing this
+     * {@code ChangeRequest}.
+     */
+    public Builder additions(List<DnsRecord> additions) {
+      this.additions = Lists.newLinkedList(checkNotNull(additions));
+      return this;
+    }
+
+    /**
+     * Adds a {@link DnsRecord} which to be <strong>added</strong> to the zone upon executing this
+     * {@code ChangeRequest}.
+     */
+    public Builder add(DnsRecord record) {
+      this.additions.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Adds a {@link DnsRecord} which to be <strong>deleted</strong> to the zone upon executing this
+     * {@code ChangeRequest}.
+     */
+    public Builder delete(DnsRecord record) {
+      this.deletions.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Clears the collection of {@link DnsRecord}s which are to be added to the zone upon executing
+     * this {@code ChangeRequest}.
+     */
+    public Builder clearAdditions() {
+      this.additions.clear();
+      return this;
+    }
+
+    /**
+     * Clears the collection of {@link DnsRecord}s which are to be deleted from the zone upon
+     * executing this {@code ChangeRequest}.
+     */
+    public Builder clearDeletions() {
+      this.deletions.clear();
+      return this;
+    }
+
+    /**
+     * Removes a single {@link DnsRecord} from the collection of of records to be
+     * <strong>added</strong> to the zone upon executing this {@code ChangeRequest}.
+     */
+    public Builder removeAddition(DnsRecord record) {
+      this.additions.remove(record);
+      return this;
+    }
+
+    /**
+     * Removes a single {@link DnsRecord} from the collection of of records to be
+     * <strong>deleted</strong> from the zone upon executing this {@code ChangeRequest}.
+     */
+    public Builder removeDeletion(DnsRecord record) {
+      this.deletions.remove(record);
+      return this;
+    }
+
+    /**
+     * Sets a collection of {@link DnsRecord}s which are to be deleted from the zone upon executing
+     * this {@code ChangeRequest}.
+     */
+    public Builder deletions(List<DnsRecord> deletions) {
+      this.deletions = Lists.newLinkedList(checkNotNull(deletions));
+      return this;
+    }
+
+    /**
+     * Associates a server-assigned id to this {@code ChangeRequest}.
```

Comment:
Acknowledged. I wrote this one specifically for myself so to remind me that this is a server-assigned value which should be read only.